irispy/iris_tools.py

```diff
@@ -212,6 +220,88 @@ def get_iris_response(pre_launch=False, response_file=None, response_version=Non
     return iris_response
 
 
+def fit_iris_xput(time_obs, time_cal_coeffs, cal_coeffs):
+    """
+    To calculate the coefficients of best-fit time function for throughput,
+    for which there are two modes:
+    1. Perform fit: supply xput and single element ``cal_coeffs``.
+    2. Apply fit: supply full ``cal_coeffs``.
+
+    Parameters
+    ----------
+    time_obs: a numpy array of floats
+    - Times of datapoints.
+
+    time_cal_coeffs: a numpy array of floats
+    - Start and end times of intervals of constant ``cal_coeffs[i]``.
+
+    cal_coeffs: a numpy array of floats
+    - Coefficients of best-fit function.
+
+    Returns
+    -------
+    Yields the values of fit for times ``time_obs``.
+
+    """
+    if isinstance(time_cal_coeffs, np.ndarray):
+        if isinstance(cal_coeffs, np.ndarray):
+            size_time_cal_coeffs = time_cal_coeffs.shape
+            size_cal_coeffs = cal_coeffs.shape
+
+    if size_time_cal_coeffs[1] != 2 or size_cal_coeffs[1] < 2:
+        # Raise ValueError as time coefficient have the wrong format.
+        raise ValueError("Incorrect number of elements either in time_cal_coeffs or in cal_coeffs.")
+    # Exponent for transition between exp.decay intervals.
+    transition_exp = 1.5
+    yr2sec = np.float64(365.25*24.*3600.)
+    fit_out = np.zeros(len(list(time_obs)), dtype=np.float64)
+
+    # Some time transformations.
+    t_cal_coeffs = np.float64(time_cal_coeffs)
+    t_ref = parse_time('1979-01-01', time=1)
+
+    count = 0
+    for i in time_obs:
+        aux_cal_coeffs = np.zeros(2*size_time_cal_coeffs[0])
+        t_obs = parse_time(i, utime=1) - t_ref
+        t_obs = t_obs.total_seconds()
```

Comment:
Since ```time_obs``` and ```t_cal_coeffs``` will have been converted to ```datetime.datetime``` the two lines above can be deleted and [L271](https://github.com/sunpy/irispy/pull/108/files#diff-dee23d5475a6ed5e368938455c92fb2eR271) can be changed to ```t_diff = (t_obs - t_cal_coeffs).total_seconds()```